### PR TITLE
Add basic doc for eventing OIDC feature

### LINKED ISSUE #5799: Add documentation for OIDC feature flag
## What is changing?

Knative Eventing will support OIDC authentication between senders and receivers. 
When the `authentication-oidc` feature flag is set to `true`, the core resources addressabled (e.g. Broker, Channel, Sequence, Parallel) will announce their OIDC audience in their status (`.status.address.audience`).
Senders (e.g. channel/broker dispatchers) will request an OIDC token for their targets audience and add it to their request (as a Bearer token in the `Authorization` header). The subject of the OIDC token, will be a dedicated service accounts identity from the sender (each sending resource from eventing-core, will create an additional OIDC service account only for this purpose).

## How will this impact our users?

When the OIDC feature flag is enabled, eventing-core addressables will reject requests which do not contain a Bearer token in their `Authorization` header and if the audience of the token does not match the addressables audience.

## In what release will this take happen (to the best of your knowledge)?
Ex. v1.13

## Context
Project dashboard: https://github.com/orgs/knative/projects/66

### PR DESCRIPTION
**Pull request by creydr**
Fixes #5799

## Proposed Changes

- :gift: Add basic documentation for eventing `authentication-oidc` feature flag

### CODE REVIEW

**Review by Cali0707 — CHANGES_REQUESTED** [review]
Hey @creydr would you be able to update https://github.com/knative/docs/blob/9767277d4e711bfaf11b768d68a928f1d402c5b9/config/nav.yml#L278-L286 so that the new page you've added is properly included in the site navigation?

**Comment by creydr** [comment]
> Hey @creydr would you be able to update
> 
> https://github.com/knative/docs/blob/9767277d4e711bfaf11b768d68a928f1d402c5b9/config/nav.yml#L278-L286
> 
> so that the new page you've added is properly included in the site navigation?

ah sure. Sorry forgot it...

File: docs/eventing/experimental-features/sender-identity.md [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+
+Currently, event delivery within the cluster is unauthenticated, and an addressable event consumer cannot determine the identity of any sender.
+
+Knative Eventing Addressables expose their OIDC Audience in their status as part of their address (e.g. `.status.address.audience`) and require requests containing an OIDC access token issued for this audience.
+
+Knative Eventing Souces request an OIDC access token for the targets audience and add them to the request. A per-source dedicated Service Account is used as the identity for the request.
+
+## Prerequisites
+
+- [Eventing installation](./../../install)
```
Comment (Leo6Leo) [inline docs/eventing/experimental-features/sender-identity.md:19]: The link here is invalid. I would suggest use the absolute path

File: docs/eventing/experimental-features/sender-identity.md [hunk 2]
```diff
@@ -0,0 +1,233 @@
...
+the following:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: config-features
+  namespace: knative-eventing
+data:
+  authentication-oidc: "enabled"
```
Comment (Leo6Leo) [inline docs/eventing/experimental-features/sender-identity.md:41]: do we want to mention that TLS should also be enabled?

**Review by Leo6Leo — CHANGES_REQUESTED** [review]

**Comment by Leo6Leo** [comment]
And tested all the sample script and yaml files in the doc, they all work as expected.

File: docs/eventing/experimental-features/sender-identity.md (hunk 2 above)
Comment (creydr) [reply docs/eventing/experimental-features/sender-identity.md:44]: Good point. I added a note about TLS too

File: docs/eventing/experimental-features/sender-identity.md (hunk 1 above)
Comment (creydr) [reply docs/eventing/experimental-features/sender-identity.md]: Good catch. Updated it

**Pull request merged**